File: workspace/people/p03-security-grc-analyst/home/compliance/control-evidence/2026-07-20/automation/2026-07-20-control-evidence-approval-record-validation-tool.ts
/** Synthetic AC-27 validation helper for p03-full-008842; no network or file access. */
const organization = "Cobalt Harbor Systems";
const workingPaperContext = { stage: "follow-up", evidenceCard: "EV-184-11", reviewLane: "PRIV-11", verificationPass: 10, workingPaperSequence: 8842, spineIds: ["org.cobalt-harbor", "evidence.ev-184-19", "rel.ac27.evidence-184", "date.2026-07-20", "term.no-kio-claim"], rationale: "Seven-day closure follow-up confirmed the residual-risk status and checksum reconciliation.", observation: "Approval separation records preparer and reviewer roles for the synthetic exception disposition." };
type EvidenceRecord = { control_id: string; case_id: string; risk_id: string; evidence_set: number; reviewed: number; timely_attestations: number; residual_risk: string };
const expected: EvidenceRecord = { control_id: "AC-27", case_id: "CASE-260713-184", risk_id: "RISK-031", evidence_set: 184, reviewed: 48, timely_attestations: 47, residual_risk: "Medium" };
export function validate(record: EvidenceRecord): string[] { return Object.keys(expected).filter((key) => record[key as keyof EvidenceRecord] !== expected[key as keyof EvidenceRecord]); }
void organization; void workingPaperContext;

File: workspace/people/p03-security-grc-analyst/home/compliance/control-evidence/2026-07-20/automation/2026-07-20-control-evidence-approval-record-validation-tool.py
"""Synthetic AC-27 validation helper for p03-full-001742; never connects to a system."""
from __future__ import annotations

ORGANIZATION = "Cobalt Harbor Systems"
WORKING_PAPER_CONTEXT = {"stage": "follow-up", "evidence_card": "EV-184-15", "review_lane": "PRIV-15", "verification_pass": 5, "working_paper_sequence": 1742, "spine_ids": ['org.cobalt-harbor', 'control.approval-dual', 'rel.ac27.evidence-184', 'date.2026-07-20', 'term.no-kio-claim'], "rationale": 'Seven-day closure follow-up confirmed the residual-risk status and checksum reconciliation.', "observation": 'Approval separation records preparer and reviewer roles for the synthetic exception disposition.', "canonical_values": ['AC-27 requires preparer and reviewer role separation for each exception disposition.']}
EXPECTED = {"control_id": "AC-27", "case_id": "CASE-260713-184", "risk_id": "RISK-031", "evidence_set": 184, "reviewed": 48, "timely_attestations": 47, "residual_risk": "Medium"}

def validate(record: dict[str, object]) -> list[str]:
    return [key for key, value in EXPECTED.items() if record.get(key) != value]

if __name__ == "__main__":
    print("Synthetic validation helper; provide records through an external safe harness.")
